test/internal_rules.jl

```diff
@@ -359,5 +360,27 @@ end
         @test isapprox(dA, dA_sym)
     end
 end
+
+@testset "Linear solve for triangular matrices" begin
+    @testset for T in (UpperTriangular, LowerTriangular, UnitUpperTriangular, UnitLowerTriangular),
+        TE in (Float64, ComplexF64), sizeB in ((3,), (3, 3))
+        M = rand(TE, 3, 3)
+        B = rand(TE, sizeB...)
+        Y = zeros(TE, sizeB...)
+        A = T(M)
```

Comment:
yeah batch tests as well would be nice

Reply:
I added some tests, but they don't pass yet. Maybe we can trigger CI to get a full picture. Will fix them later.